diff -Naur original/lib/zip_source_filep.c libzip-1.0.1/lib/zip_source_filep.c
--- original/lib/zip_source_filep.c	2015-05-04 17:49:13
+++ libzip-1.0.1/lib/zip_source_filep.c	2024-08-19 12:53:37
@@ -183,14 +183,14 @@
     }
     sprintf(temp, "%s.XXXXXX", ctx->fname);
 
-    mask = umask(S_IXUSR | S_IRWXG | S_IRWXO);
+//    mask = umask(S_IXUSR | S_IRWXG | S_IRWXO);
     if ((tfd=mkstemp(temp)) == -1) {
         zip_error_set(&ctx->error, ZIP_ER_TMPOPEN, errno);
-	umask(mask);
+//	umask(mask);
         free(temp);
         return -1;
     }
-    umask(mask);
+//    umask(mask);
 
     if ((tfp=fdopen(tfd, "r+b")) == NULL) {
         zip_error_set(&ctx->error, ZIP_ER_TMPOPEN, errno);
@@ -246,10 +246,10 @@
                 zip_error_set(&ctx->error, ZIP_ER_RENAME, errno);
                 return -1;
             }
-	    mask = umask(022);
-	    umask(mask);
+//	    mask = umask(022);
+//	    umask(mask);
 	    /* not much we can do if chmod fails except make the whole commit fail */
-	    (void)chmod(ctx->fname, 0666&~mask);
+//	    (void)chmod(ctx->fname, 0666&~mask);
 	    free(ctx->tmpname);
 	    ctx->tmpname = NULL;
             return 0;
